src/components/decimal/decimal.component.js

```diff
@@ -11,247 +16,261 @@ import LocaleContext from "../../__internal__/i18n-context";
+      );
+      return valueToFormat.replace(delimiterMatcher, "");
+    },
+    [getSeparator]
+  );
 
   /**
    * Convert raw input to a standard decimal format
    */
-  toStandardDecimal = (i18nValue) => {
-    const valueWithoutNBS =
-      this.getSeparator("group").match(/\s+/) && !i18nValue.match(/\s{2,}/)
-        ? i18nValue.replace(/\s+/g, "")
-        : i18nValue;
-    /* If a value is passed in that is a number but has too many delimiters in succession, we want to handle this
+  const toStandardDecimal = useCallback(
+    (i18nValue) => {
+      const valueWithoutNBS =
+        getSeparator("group").match(/\s+/) && !i18nValue.match(/\s{2,}/)
+          ? i18nValue.replace(/\s+/g, "")
+          : i18nValue;
+      /* If a value is passed in that is a number but has too many delimiters in succession, we want to handle this
     value without formatting it or removing delimiters. We also want to consider that,
     if a value consists of only delimiters, we want to treat that 
     value in the same way as if the value was NaN. We want to pass this value to the 
     formatValue function, before the delimiters can be removed. */
-    const errorsWithDelimiter = new RegExp(
-      `([^A-Za-z0-9]{2,})|(^[^A-Za-z0-9-]+)|([^0-9a-z-,.])|([^0-9-,.]+)|([W,.])$`,
-      "g"
-    );
-    const separator = this.getSeparator("decimal");
-    const separatorRegex = new RegExp(
-      separator === "." ? `\\${separator}` : separator,
-      "g"
-    );
-    if (
-      valueWithoutNBS.match(errorsWithDelimiter) ||
-      (valueWithoutNBS.match(separatorRegex) || []).length > 1
-    ) {
-      return valueWithoutNBS;
+      const errorsWithDelimiter = new RegExp(
+        `([^A-Za-z0-9]{2,})|(^[^A-Za-z0-9-]+)|([^0-9a-z-,.])|([^0-9-,.]+)|([W,.])$`,
+        "g"
+      );
+      const separator = getSeparator("decimal");
+      const separatorRegex = new RegExp(
+        separator === "." ? `\\${separator}` : separator,
+        "g"
+      );
+      if (
+        valueWithoutNBS.match(errorsWithDelimiter) ||
+        (valueWithoutNBS.match(separatorRegex) || []).length > 1
+      ) {
+        return valueWithoutNBS;
+      }
+
+      const withoutDelimiters = removeDelimiters(valueWithoutNBS);
+      return withoutDelimiters.replace(new RegExp(`\\${separator}`, "g"), ".");
+    },
+    [getSeparator, removeDelimiters]
+  );
+
+  const decimalValue = getSafeValueProp(defaultValue || value || emptyValue);
+  const [stateValue, setStateValue] = useState(
+    isNaN(toStandardDecimal(decimalValue))
+      ? decimalValue
+      : formatValue(decimalValue)
+  );
+
+  const createEvent = useCallback(
+    (formatted, raw) => {
+      return {
+        target: {
+          name,
+          id,
+          value: {
+            formattedValue: formatValue(toStandardDecimal(formatted)),
+            rawValue: raw || toStandardDecimal(formatted),
+          },
+        },
+      };
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
You should remove this disable comment and add all the dependencies eslint tells you to here. Or do the above and remove the `useCallback` completely

Reply:
I've chosen to remove the `useCallback`. `createEvent` is in any dependancy arrays so there is no real need for it to be a `useCallback`.